app/src/androidTest/java/net/osmtracker/layouts/DownloadLayoutTest.java

```diff
@@ -0,0 +1,83 @@
+package net.osmtracker.layouts;
+
+import android.support.test.espresso.Espresso;
+import android.support.test.espresso.ViewAssertion;
+import android.support.test.espresso.action.ViewActions;
+import android.support.test.espresso.assertion.ViewAssertions;
+import android.support.test.rule.ActivityTestRule;
+
+import net.osmtracker.R;
+import net.osmtracker.activity.TrackManager;
+import net.osmtracker.util.TestUtils;
+
+import org.apache.commons.io.FileUtils;
+import org.junit.Rule;
+import org.junit.Test;
+
+import static android.support.test.InstrumentationRegistry.getInstrumentation;
+import static android.support.test.espresso.Espresso.onData;
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.Espresso.openActionBarOverflowOrOptionsMenu;
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.action.ViewActions.scrollTo;
+import static android.support.test.espresso.matcher.PreferenceMatchers.withTitleText;
+import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static android.support.test.espresso.matcher.ViewMatchers.withText;
+
+public class DownloadLayoutTest {
+    @Rule
+    public ActivityTestRule<TrackManager> mRule = new ActivityTestRule<>(TrackManager.class);
+
+    public void deleteLayoutsDirectory(){
+        try {
+            FileUtils.deleteDirectory(TestUtils.getLayoutsDirectory());
+        }catch (Exception e){
+            e.printStackTrace();
+        }
+    }
+
+
+    @Test
+    public void downloadLayoutTest() {
```

Comment:
This test is OK as it is now. 
As a improvement I consider will be better if we create a "Test Layout" in the [layouts repo](https://github.com/labexp/osmtracker-android-layouts) only for testing porposes. Because now we are using an active custom layout (hidrantes) and I guess maybe the text shown in the layout could change and that will cause a collateral affect on the test.

Reply:
That's true, I'll create a specific branch for testing purposes to avoid using the production one for that.